src/composition/supergraph/config/mod.rs

```diff
@@ -298,10 +298,33 @@ impl FinalSupergraphConfig {
         Ok(())
     }
 
-    pub fn federation_version(&self) -> FederationVersion {
-        self.config
-            .get_federation_version()
-            .unwrap_or(FederationVersion::LatestFedTwo)
+    /// Calculates what the correct version of Federation should be, based on the
+    /// value on the given environment variable or the supergraph config.
+    ///
+    /// The order of precedence is:
+    /// Environment Variable -> Supergraph Schema -> Default (Latest)
+    pub fn federation_version(&self, env_var: Option<String>) -> FederationVersion {
+        let env_var_version = if let Some(version) = env_var {
+            match FederationVersion::from_str(&version) {
```

Comment:
I copied this from [here](https://github.com/apollographql/rover/blob/loshz/ROVER-174/src/command/dev/protocol/leader.rs#L163), but I removed the exact version constraint.